client/src/components/forum/ForumMain.jsx

```diff
@@ -1,10 +1,82 @@
 import React, { useState, useEffect } from 'react';
-import './ForumMain.css';
-import ForumSubmission from './ForumSubmissionForm.jsx';
+import './ForumMain.css'; 
+import ForumSubmissionForm from './ForumSubmissionForm.jsx';
 import NavbarComp from '../navbar/Navbar.jsx';
+import ThreadReply from './ThreadReply.jsx'; // Import the ThreadReply component
 import { Link } from 'react-router-dom';
 
 const ForumMain = () => {
+  const [threads, setThreads] = useState([]);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    fetch('http://localhost:3000/threads')
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then(data => {
+        setThreads(data);
+      })
+      .catch(error => {
+        console.error('Error fetching threads:', error);
+        setError('Failed to fetch threads');
+      });
+  }, []);
+
+  const handleReplySubmit = (newReply) => {
```

Comment:
All of this functionality for the Replies needs to be added into SingleThread.jsx. 

Once that's done pls revert the forum main code back to what is on development currently as you've removed a lot of the functionality and it's no longer working as intended, but just ensure you save your form code so it can be put in :o)